### PR TITLE
Paradigm Restructuring Part 2 - Storage

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp #10918 
Addons: https://github.com/friendica/friendica-addons/pull/1196

- Storage restructuring

### CODE REVIEW

File: src/Console/Storage.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ class Storage extends \Asika\SimpleConsole\Console
 {
 	protected $helpOptions = ['h', 'help', '?'];
 
-	/** @var StorageManager */
+	/** @var \Friendica\Core\Storage\Repository\StorageManager */
```
Comment (MrPetovan) [inline src/Console/Storage.php]: Extraneous absolute path
```suggestion
	/** @var StorageManager */
```

File: src/Core/Storage/Capability/ICanReadFromStore.php [hunk 2]
```diff
@@ -19,12 +19,15 @@
...
+namespace Friendica\Core\Storage\Capability;
+
+use Friendica\Core\Storage\Exception\ReferenceStorageException;
+use Friendica\Core\Storage\Exception\StorageException;
 
 /**
  * Interface for basic storage backends
  */
-interface IStorage
+interface ICanReadFromStore
```
Comment (MrPetovan) [inline src/Core/Storage/Capability/ICanReadFromStore.php]: This is so much better!

File: src/Core/Storage/Repository/StorageManager.php [hunk 3]
```diff
@@ -385,7 +391,7 @@ public function move(Storage\IWritableStorage $destination, array $tables = self
 					$this->logger->info('update row');
 					if ($this->dba->update($table, ['backend-class' => $destination::getName(), 'backend-ref' => $destinationRef, 'data' => ''], ['id' => $id])) {
 						if (!empty($source)) {
-							$this->logger->info('Delete data from old backend.', ['oldBackend' => $source, 'oldReference' => $sourceRef]);
+							$this->logger->info('Deleted data from old backend.', ['oldBackend' => $source, 'oldReference' => $sourceRef]);
```
Comment (MrPetovan) [inline src/Core/Storage/Repository/StorageManager.php:394]: Why the change of tense?

File: src/Module/Admin/Storage.php [hunk 4]
```diff
@@ -40,7 +40,7 @@ public static function post(array $parameters = [])
 		$storagebackend = Strings::escapeTags(trim($parameters['name'] ?? ''));
 
 		try {
-			/** @var IStorageConfiguration|false $newStorageConfig */
+			/** @var \Friendica\Core\Storage\Capability\ICanConfigureStorage|false $newStorageConfig */
```
Comment (MrPetovan) [inline src/Module/Admin/Storage.php]: Extraneous absolute name:
```suggestion
			/** @var ICanConfigureStorage|false $newStorageConfig */
```

File: src/Module/Admin/Storage.php [hunk 5]
```diff
@@ -78,7 +78,7 @@ public static function post(array $parameters = [])
 
 		if (!empty($_POST['submit_save_set'])) {
 			try {
-				/** @var IWritableStorage $newstorage */
+				/** @var \Friendica\Core\Storage\Capability\ICanWriteToStorage $newstorage */
```
Comment (MrPetovan) [inline src/Module/Admin/Storage.php]: Extraneous absolute name:
```suggestion
				/** @var ICanWriteToStorage $newstorage */
```

File: tests/src/Core/Storage/StorageConfigTest.php [hunk 6]
```diff
@@ -19,17 +19,17 @@
...
+namespace Friendica\Test\src\Core\Storage;
 
-use Friendica\Model\Storage\IStorageConfiguration;
+use Friendica\Core\Storage\Capability\ICanConfigureStorage;
 use Friendica\Test\MockedTest;
 
 abstract class StorageConfigTest extends MockedTest
 {
-	/** @return IStorageConfiguration */
+	/** @return \Friendica\Core\Storage\Capability\ICanConfigureStorage */
```
Comment (MrPetovan) [inline tests/src/Core/Storage/StorageConfigTest.php]: Extraneous absolute name:
```suggestion
	/** @return ICanConfigureStorage */
```

File: tests/src/Core/Storage/StorageConfigTest.php [hunk 7]
```diff
@@ -19,17 +19,17 @@
...
 	abstract protected function getInstance();
 
-	abstract protected function assertOption(IStorageConfiguration $storage);
+	abstract protected function assertOption(\Friendica\Core\Storage\Capability\ICanConfigureStorage $storage);
```
Comment (MrPetovan) [inline tests/src/Core/Storage/StorageConfigTest.php]: Extraneous absolute name:
```suggestion
	abstract protected function assertOption(ICanConfigureStorage $storage);
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Core/Storage/Repository/StorageManager.php (hunk 3 above)
Comment (nupplaphil) [reply src/Core/Storage/Repository/StorageManager.php:394]: tbh .. My PHPStorm all the time complains about the non-conform SQL-language and I don't find a switch to ignore it in this case :D

**Review by MrPetovan — APPROVED** [review]

**Comment by nupplaphil** [comment]
looks good, my node works (even with webdav addon :) )

**Pull request merged**